backend/controllers/jobPostingsController.js

```diff
@@ -0,0 +1,104 @@
+const { mongo, default: mongoose } = require('mongoose');
+const JobPosting = require('../models/jobPostingModel');
+
+// get all job postings
+const getJobPostings = async (req, res) => {
+    const jobPostings = await JobPosting.find({});
+
+    res.status(200).json(jobPostings);
+}
+
+// get a job posting by ID
+const getJobPostingById = async (req, res) => {
+    const { id } = req.params
+
+    if (!mongoose.Types.ObjectId.isValid(id)) {
+        return res.status(404).send('Job posting not found')
+    }
+
+    const jobPosting = await JobPosting.findById(id);
+
+    if (!jobPosting) {
+        return res.status(404).send('Job posting not found')
+    }
+
+    res.status(200).json(jobPosting);
+}
+
+// create a new job posting
+const createJobPosting = async (req, res) => {
+    const { company, jobTitle, location, jobType, details, noDegreeMentioned, benefits } = req.body;
+
+    const newJobPosting = new JobPosting({
```

Comment:
Should move this after the logic for checking if the same job already exists. Also shouldnt it be possible for two jobs to have the same title?